meshtastic/interfaces/ble/interface.py

```
@@ -1268,6 +1282,164 @@ def find_device(self, address: str | None) -> BLEDevice:
+            return existing_client_address
+        if requested_identifier and _looks_like_ble_address(requested_identifier):
+            return requested_identifier
+        return self.findDevice(requested_identifier).address
+
+    def _format_bluetoothctl_address(self, address: str) -> str:
+        """Canonicalize a BLE address for bluetoothctl commands."""
+        sanitized = sanitize_address(address)
+        if sanitized is None or len(sanitized) != 12:
+            raise self.BLEError(
+                f"Cannot trust device: {address!r} did not resolve to a BLE address."
+            )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)